### ISSUE TITLE
New stalker version being able to read old stalker database

### ISSUE THREAD

**Issue by fredrikaverpil** [eoyilmaz/stalker#53]
I noticed that I was able to read a version 0.2.15.1 Stalker database using stalker 0.2.18.

Would it be possible to tell Stalker to abort the database connection if the (alembic?) versions doesn't match?

**Comment by jasperges** [comment]
Mmm, that's weird. I just wanted to read the database (created in Stalker 0.2.17.6) in version 02.18 and got an error. I checked again with opening a database created with Stalker 0.2.15.1 (just to be sure) and got the same error. This is when using `db.setup()`. So it seems to do exactly what you would like it to do. Strange that you didn't get this. This is the traceback I got:

```python
----> 1 db.setup()

/Volumes/data/src/project_management/stalker/stalker/db/__init__.py in setup(settings)
     70     # check alembic versions of the database
     71     # and raise an error if it is not matching with the system
---> 72     check_alembic_version()
     73
     74     # create the database

/Volumes/data/src/project_management/stalker/stalker/db/__init__.py in check_alembic_version()
    213     if current_alembic_version and current_alembic_version != alembic_version:
    214         raise ValueError(
--> 215             'Please update the database to version: %s' % alembic_version
    216         )
    217

ValueError: Please update the database to version: 31b1e22b455e
```

**Comment by eoyilmaz** [comment]
@fredrikaverpil what is the ``alembic_version`` of the database that you are trying to connect with Stalker 0.2.18.

**Comment by fredrikaverpil** [comment]
I don't have access to the database until maybe tomorrow, as I'm working from home due to an Internet outage at our studio. But looking at #51, it should be `f2005d1fbadc`.

**Comment by eoyilmaz** [comment]
Is this issue still present, can we close it or does it need further investigation.

**Comment by fredrikaverpil** [comment]
I can check today if 0.2.20 can read a 0.2.18 database and will report back.

**Comment by fredrikaverpil** [comment]
This issue can be close! 👍 😄 

When attempting to connect to a 0.2.18 db with Stalker 0.2.20, I get this:

    Please update the database to version: ed0167fff399

**Comment by eoyilmaz** [comment]
Thanks 👍

**Comment by eoyilmaz** [comment]
I found that upon connection to an older database, even that Stalker is raising a ValueError (where it should be raising a database related error) the session remains open, so this is issue is not closed yet. I'll commit a fix soon.

**Comment by eoyilmaz** [comment]
Fixed in  ef21f1b
